tests/bdd/features/cordon/node/test_cordon_node.py

```diff
@@ -0,0 +1,451 @@
+"""Cordoning feature tests."""
+
+from pytest_bdd import (
+    given,
+    scenario,
+    then,
+    when,
+)
+
+import os
+import pytest
+import requests
+
+from common.apiclient import ApiClient
+from common.operations import Cluster
+from common.deployer import Deployer
+from common.docker import Docker
+
+from openapi.model.create_pool_body import CreatePoolBody
+from openapi.model.create_volume_body import CreateVolumeBody
+from openapi.model.volume_policy import VolumePolicy
+
+VOLUME_UUID_1 = "5cd5378e-3f05-47f1-a830-111111111111"
+VOLUME_UUID_2 = "5cd5378e-3f05-47f1-a830-222222222222"
+
+VOLUME_SIZE = 1048
+NUM_VOLUME_REPLICAS = 1
+CREATE_REQUEST_KEY = "create_request"
+
+NODE_NAME_1 = "io-engine-1"
+NODE_NAME_2 = "io-engine-2"
+
+POOL_UUID_1 = "4cc6ee64-7232-497d-a26f-111111111111"
+
+CORDON_LABEL_1 = "cordon_label_1"
+CORDON_LABEL_2 = "cordon_label_2"
+
+
+def pytest_configure():
+    pytest.command_failed = False
+
+
+@pytest.fixture(scope="module")
+def init():
+    Deployer.start(2)
+    yield
+    Deployer.stop()
+
+
+@pytest.fixture(autouse=True)
+def init_scenario(init, disks):
+    pytest.command_failed = False
+    ApiClient.pools_api().put_node_pool(
+        NODE_NAME_1, POOL_UUID_1, CreatePoolBody([f"{disks[0]}"])
+    )
+    yield
+    Docker.restart_container("core")
+    Cluster.cleanup()
+    remove_all_cordons(NODE_NAME_1)
+    remove_all_cordons(NODE_NAME_2)
+    delete_all_volumes()
+
+
+@pytest.fixture
+def tmp_files():
+    files = []
+    for index in range(0, 1):
+        files.append(f"/tmp/disk_{index}")
+    yield files
+
+
+@pytest.fixture
+def disks(tmp_files):
+    for disk in tmp_files:
+        if os.path.exists(disk):
+            os.remove(disk)
+        with open(disk, "w") as file:
+            file.truncate(100 * 1024 * 1024)
+    # /tmp is mapped into /host/tmp within the io-engine containers
+    yield list(map(lambda file: f"/host{file}", tmp_files))
+
+    for disk in tmp_files:
+        if os.path.exists(disk):
+            os.remove(disk)
+
+
+# Fixture used to pass the volume create request between test steps.
+@pytest.fixture(scope="function")
+def create_request():
+    return {}
+
+
+###### utility functions ######
+
+
+def cordon_node(node_name, label):
+    ApiClient.nodes_api().put_node_cordon(node_name, label)
+    assert is_cordoned(node_name) == True
+
+
+def create_volume(vol_uuid):
+    request = CreateVolumeBody(
+        VolumePolicy(False), NUM_VOLUME_REPLICAS, VOLUME_SIZE, False
+    )
+    ApiClient.volumes_api().put_volume(vol_uuid, request)
+    # Check that the volume was created.
+    volumes = ApiClient.volumes_api().get_volume(vol_uuid)
+
+
+def create_volume_and_fail(vol_uuid):
+    request = CreateVolumeBody(
+        VolumePolicy(False), NUM_VOLUME_REPLICAS, VOLUME_SIZE, False
+    )
+    try:
+        ApiClient.volumes_api().put_volume(vol_uuid, request)
+        assert False
+    except Exception as e:
+        print("failed, as expected")
```

Comment:
nit: I'd suggest either adding more details to these prints or replacing them with `pass`

Reply:
Added more details to the message.